### DISCUSSION TITLE
TS71002: "API_URL" is not a valid Next.js entry export value.

### DISCUSSION THREAD

**Discussion by HRCEO** [vercel/next.js#61619] [Help]
export const API_URL: string = 'https://nomad-movies.nomadcoders.workers.dev/movies';

    async function getMovies() {
        await new Promise((resolve)=>setTimeout(resolve, 1000))
        const response = await fetch(API_URL)
        const json = await response.json()
        return json;
    }

    export default async function HomePage(){ ... } 
---

    import {API_URL} from "../../../(home)/page";

    async function getMovie(id:string){
        const response = await fetch(`${API_URL}/${id}`)
        return response.json()
    }

    export default async function MovieDetail({params:{id}})
    {
        const movie = await getMovie(id)
        return <h1>{movie.title}</h1>
    }

error : `TS71002: 'API_URL' is not a valid Next.js entry export value.` (HomePage.tsx file)

I declare a variable in the HomePage component and export it to be called and used in the movie. Functionally, it works fine without any issues, but a type error occurs. Why is that?

---
tsconfig.json

    {
      "compilerOptions": {
        "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "noEmit": true,
    "incremental": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
        "jsx": "preserve",
        "plugins": [
          {
            "name": "next"
          }
        ]
      },
      "include": [
        "next-env.d.ts",
        ".next/types/**/*.ts",
        "**/*.ts",
        "**/*.tsx"
      ],
      "exclude": [
        "node_modules"
      ]
    }

**Comment by makibat** [comment]
Have you tried restarting the TS server in VS Code?
